README.md

```diff
@@ -1 +1,105 @@
+
+This text should be kept short, to the point, and as unambiguous as possible. Define any project-specific jargon, avoid editorializing, and limit subtle language. Use diagrams and imagery where appropriate.
+
+# Raison d'Être
+
+> Had the [Web] been proprietary, and in my total control, it would probably not have taken off. You can’t propose that something be a universal space and at the same time keep control of it.
+>
+> — Tim Berners-Lee, [FAQ][TBL FAQ]
+
+IPVM intends to do nothing less than connect all of the world's users and services. It can be thought of as "the HTTP of compute": open, interoperable, and everywhere. Following the model of the Web being the network of linked documents, IPVM is the network of linked computation. It allows anyone to permissionlessly tie into the network without prenegotiation.
+
+TCP/IP and HTTP were successful for a number of reasons. Networking was always possible, but doing this from scratch for every application is infeasible. Hypertext provided a single mechanism for many applications to interconnect. Application agnosticism is a major pillar of [such an approach][W3F Principles]: anything can run on this substrate, and it's easy to join. IPVM works equally well with traditional Cloud-based services as it does local-first compute and trustless architectures. To be successful, IPVM must have a path towards becoming substantially better than a pure Cloud architecture. It does not stand in opposition to the Cloud, but rather extends and contains it.
+
+The internet is no longer purely client/server. Consumer devices are now significantly more powerful than when the LAMP stack and Cloud infrastructure were being co-developed. Devices today are heterogeneous: there are powerful browsers, heavy servers, microservices, edge PoPs, tiny IoT devices, smartphones, and commons networks like BitTorrent, IPFS, and blockchains. IPVM provides a way to tie applications together with robust invocation, routing, and trust layers. The volume of data that modern applications requires leads to a situation known as data gravity. Given the speed of light, the amount of data that we produce and modify grows faster than is possible to sync to all locations, and .
+
+# Audience
+
+IPVM is intended for general distributed computation, but excels at tasks that would be found in systems like Lambda Step functions. The IPVM network itself is focused on machine-to-machine interaction — where each node acts as the user's agent — but is not expected to be interacted with directly when on the happy path.
+
+# Goals
+
+IPVM means to provide:
+
+- Task matchmaking network / marketplace
+- Pluggable execution engine
```

Comment:
Admittedly, I still think we should lean more toward a sidecar/agent approach than being directly pluggable, but I'm not sure that's worth the verbiage?

Reply:
Clearly I should reword this: I mean the effect plugin system that runs on Wasm components

Reply:
But while we're on the topic, why not write the code in a way where we expose the Homestar internals as crates that can be recomposed in other apps?

Reply:
UPDATE: I just removed this bullet point. It's basically covered by two later in the list

Reply:
@expede For wasm/wasi components (to the intention), yeah, for sure. And, yeah, any internals that can be generally used will be exposed as crates without a doubt. When I read the plugin initially, I viewed it as more general. I do think there are methods to incorporate "3rd-party" execution that does not involve wrapping everything within a wasm context. But, I do agree that we want to more easily support wasm/wasi plugins as an obvious primitive.